src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -581,7 +581,8 @@ private void ProcessPingFrame(FrameHeader frameHeader)
 
             // Send PING ACK
             // Don't wait for completion, which could happen asynchronously.
-            LogExceptions(SendPingAckAsync(_incomingBuffer.ActiveMemory.Slice(0, FrameHeader.PingLength)));
+            byte[] pingContent = _incomingBuffer.ActiveMemory.Slice(0, FrameHeader.PingLength).ToArray();
```

Comment:
I think we could just do BinaryPrimitives.ReadInt64BigEndian or whatever it is, and avoid an alloc here.

Reply:
I considered it, but the rarity of PING makes me feel that this is not a worthwhile micro-optimization given that we'd then need to document endianness.

Reply:
> we'd then need to document endianness.

Why? Wouldn't this just be roundtripping the data? As long as you used routines for converting to and from a long that employed the same endianness, it wouldn't be visible. You can also use BitConverter.ToInt64/TryWriteBytes, which just uses the machine's endianness and won't force any reversals.

Reply:
When it doesn't matter, like in this case, I'd rather reduce complexities and have obvious code than need to comment something.

> As long as you used routines for converting to and from a long that employed the same endianness, it wouldn't be visible. You can also use BitConverter.

Either way, this is going to need comments. If you know HTTP2 and know these are just opaque bytes and not an integer, you'll find it suspicious to see an endian-specific read. Even if you didn't know HTTP2, if you see `BitConverter` being used to read an integer in networking code you'd immediately flag it as a possible bug.

Reply:
> Either way, this is going to need comments

It needs comments even with ToArray.  Someone reviewing the code in a year needs to know that this wasn't an oversight and what race conditions it's working around. Otherwise it's very likely to be reverted by someone trying to remove allocations.

> When it doesn't matter, like in this case, I'd rather reduce complexities and have obvious code

It shouldn't be any more, or any more complicated, code. It's ToInt64 instead of ToArray, and TryWriteBytes instead of CopyTo. If anything it's arguably clearer code, because it makes the payload size crystal clear in the method contract, whereas today it appears to accept arbitrarily-sized data.

I assume pings are super uncommon, so from a perf perspective it probably doesn't matter. But at least from my perspective the code would be cleaner with a long, and so may as well get the perf benefit, too.

Up to you. If you choose not to switch, please do add a detailed comment about why ToArray is necessary.

Reply:
We already read a bunch of stuff out of frames as big endian, so I don't see that adding one more here is much different.